recipes/gsoap/all/conanfile.py

```diff
@@ -16,28 +19,52 @@ class GsoapConan(ConanFile):
     topics = ("gsoap", "logging")
     settings = "os", "arch", "compiler", "build_type"
     options = {
+        "fPIC": [True, False],
         "with_openssl": [True, False],
         "with_ipv6": [True, False],
         "with_cookies": [True, False],
         "with_c_locale": [True, False],
     }
     default_options = {
-        'with_openssl': True,
-        'with_ipv6': True,
-        'with_cookies': True,
-        'with_c_locale': True,
+        "fPIC": True,
+        "with_openssl": True,
+        "with_ipv6": True,
+        "with_cookies": True,
+        "with_c_locale": True,
     }
+
+    exports_sources = "CMakeLists.txt", "cmake/*.cmake"
     short_paths = True
 
-    def export_sources(self):
-        tools.files.copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
-        tools.files.copy(self, "*.cmake", os.path.join(self.recipe_folder, "src"), os.path.join(self.export_sources_folder, "src"))
-        for p in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.files.copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1q")
+            self.requires("zlib/1.2.12")
+
+    def build_requirements(self):
+        if cross_building(self, skip_x64_x86=True) and hasattr(self, "settings_build"):
+            self.tool_requires("gsoap/{}".format(self.version))
+
+        if is_msvc(self):
```

Comment:
```suggestion
        if is_msvc(self):
```

In case of 2 profiles, how to check compiler of build profile with such helper function?

Reply:
Well, looking at conan source code, it's not possible. I've opened https://github.com/conan-io/conan/issues/11926